filebeat/prospector/prospector.go

```diff
@@ -1,150 +1,39 @@
-package prospector
-
-import (
-	"fmt"
-	"sync"
-	"time"
-
-	"github.com/mitchellh/hashstructure"
-
-	"github.com/elastic/beats/filebeat/channel"
-	"github.com/elastic/beats/filebeat/input/file"
-	"github.com/elastic/beats/libbeat/common"
-	"github.com/elastic/beats/libbeat/logp"
-)
-
-// Prospectorer is the interface common to all prospectors
-type Prospectorer interface {
-	Run()
-	Stop()
-	Wait()
-}
-
-// Prospector contains the prospector
-type Prospector struct {
-	config       prospectorConfig
-	prospectorer Prospectorer
-	done         chan struct{}
-	wg           *sync.WaitGroup
-	ID           uint64
-	Once         bool
-	beatDone     chan struct{}
-}
-
-// NewProspector instantiates a new prospector
-func New(
-	conf *common.Config,
-	outlet channel.Factory,
-	beatDone chan struct{},
-	states []file.State,
-	dynFields *common.MapStrPointer,
-) (*Prospector, error) {
-	prospector := &Prospector{
-		config:   defaultConfig,
-		wg:       &sync.WaitGroup{},
-		done:     make(chan struct{}),
-		Once:     false,
-		beatDone: beatDone,
-	}
+//+build go1.9
```

Comment:
Do you really want to add this build flag as we already require Golang 1.9 for beats. Not having it would make us detect if by accident something is built previous to 1.9 and have an error.

Reply:
We do require golang 1.9, but after looking at it I am not sure we enforce it the correct way.


```
 go version
go version go1.8.3 darwin/amd64
```

## This is with the pragma `build go1.9`
```
 make
go build -i -ldflags "-X github.com/elastic/beats/libbeat/version.buildTime=2018-01-16T14:18:24Z -X github.com/elastic/beats/libbeat/version.commit=c2b149f7d67b082b20f8201db719ba1c01755683"
fileset/factory.go:7:2: no buildable Go source files in /Users/ph/go/src/github.com/elastic/beats/filebeat/prospector
make: *** [filebeat] Error 1
```

## This is without the pragma `build go1.9`
```
go build -i -ldflags "-X github.com/elastic/beats/libbeat/version.buildTime=2018-01-16T14:18:42Z -X github.com/elastic/beats/libbeat/version.commit=c2b149f7d67b082b20f8201db719ba1c01755683"
# github.com/elastic/beats/libbeat/logp
../libbeat/logp/logger.go:8: syntax error: unexpected = in type declaration
make: *** [filebeat] Error 2
```